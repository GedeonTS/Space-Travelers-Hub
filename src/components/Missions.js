import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { joinMission, fetchMissions } from '../redux/missions/missions';

const bgVariant = {
  backgroundColor: '#D3D3D3',
};

const bgWhite = {
  backgroundColor: '#fff',
};

const Missions = () => {
  const missions = useSelector((state) => state.missions);

  const dispatch = useDispatch();
  const handleJoinMission = (id) => {
    dispatch(joinMission(id));
  };

  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  return (
    <table className="missions-table">
      <thead>
        <tr>
          <th className="mission-title">Mission</th>
          <th className="mission-description">Description</th>
          <th className="mission-status">status</th>
          <th className="mission-status"> </th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr
            key={mission.mission_id}
            style={missions.indexOf(mission) % 2 === 0 ? bgVariant : bgWhite}
          >
            <td>{mission.name}</td>
            <td>{mission.description}</td>
            <td>
              <p className="membership-btn">
                {!mission.reserved && 'NOT A MEMBER'}
                {mission.reserved && 'ACTIVE MEMBER'}
              </p>
            </td>
            <td>
              {!mission.reserved && (
              <button
                type="button"
                onClick={() => {
                  handleJoinMission(mission.id);
                }}
                className="join-btn"
              >
                join mission
              </button>
              )}
              {mission.reserved && (
              <button
                type="button"
                onClick={() => { handleJoinMission(mission.id); }}
                className="join-btn"
              >
                leave mission
              </button>
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Missions;
