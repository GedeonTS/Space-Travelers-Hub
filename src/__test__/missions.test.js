import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configStore';
import Missions from '../components/Missions';

describe('display missions component', () => {
  test('test if mission component renders in the dom', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(screen.queryByText(/Reserve Rockets/)).toBeNull();
  });
  test('snapshot renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
