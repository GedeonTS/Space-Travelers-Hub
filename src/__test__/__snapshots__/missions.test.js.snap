// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display missions component snapshot renders properly 1`] = `
<table
  className="missions-table"
>
  <thead>
    <tr>
      <th
        className="mission-title"
      >
        Mission
      </th>
      <th
        className="mission-description"
      >
        Description
      </th>
      <th
        className="mission-status"
      >
        status
      </th>
      <th
        className="mission-status"
      >
         
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
